opennms-webapp/src/main/java/org/opennms/web/navigate/RoleBasedNavBarEntry.java

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2022 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2022 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.web.navigate;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import com.google.common.base.Strings;
+
+public class RoleBasedNavBarEntry extends LocationBasedNavBarEntry {
+    /** comma-separated list of roles */
+    private String roles;
+
+    public String getRoles() {
+        return this.roles;
+    }
+
+    public void setRoles(String roles) {
+        this.roles = roles;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public DisplayStatus evaluate(MenuContext context) {
+        if (!Strings.isNullOrEmpty(this.roles)) {
+            List<String> roleList = Arrays.stream(this.roles.split(",")).map(String::trim).collect(Collectors.toList());
```

Comment:
It probably doesn't really matter from a performance POV, but is there any point in bothering to store this role list, only to turn it back into a stream and compare it one time?

Reply:
Should probably just be: 
```
boolean anyMatch = Arrays.stream(this.roles.split(",")).map(String::trim).anyMatch(context::isUserInRole)
```
Would that be ok?